lib/noise-wrap.js

```diff
@@ -22,26 +23,24 @@ module.exports = class NoiseWrap {
   }
 
   recv (buf) {
-    let payload = null
-
-    try {
-      payload = c.decode(messages.noisePayload, this.handshake.recv(buf))
-    } catch {
-      return null
-    }
-
+    const payload = c.decode(messages.noisePayload, this.handshake.recv(buf))
     this.remotePublicKey = toBuffer(this.handshake.rs)
     return payload
   }
 
   final () {
     if (!this.handshake.complete) throw new Error('Handshake did not finish')
 
+    const holepunchSecret = Buffer.allocUnsafe(32)
+
+    sodium.crypto_generichash(holepunchSecret, this.handshake.hash, NS.PEER_HOLEPUNCH)
+
```

Comment:
just a mental note that we might want to have a static function or someway of generating this outside the module now we need it returned from final :)